.github/workflows/build-and-release.yml

```
@@ -0,0 +1,65 @@
+name: Build and Release
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        go-version: [1.15, 1.16]
+        platform: [amd64, arm64]
+
+    steps:
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: ${{ matrix.go-version }}
+
+    - name: Check out code
+      uses: actions/checkout@v2
+
+    - name: Build
+      run: |
+        GOOS=linux GOARCH=${{ matrix.platform }} go build -o myapp-agent ./agent/cmd/runner
+
+    - name: Upload artifact
+      uses: actions/upload-artifact@v2
+      with:
+        name: myapp-agent-${{ matrix.platform }}
+        path: myapp-agent
+
+  release:
+    runs-on: ubuntu-latest
+    needs: build
+
+    steps:
+    - name: Download artifact
+      uses: actions/download-artifact@v2
+      with:
+        name: myapp-agent-${{ matrix.platform }}
+
+    - name: Create Release
+      uses: actions/create-release@v1
```

from PerformanceAnalyst
リリースジョブでのアーティファクトのダウンロードは、プラットフォームごとに行う必要があります。現在の設定では、マトリックスのプラットフォームに対応していない可能性があります。

```suggestion
name: myapp-agent-${{ matrix.platform }}
```